src/modules/85-iac/components/IacResourceStackWizard/StackProvisionerTypeStep.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useMemo } from 'react'
+import { Button, ButtonVariation, Formik, FormikForm, IconName, Layout, Text, ThumbnailSelect } from '@harness/uicore'
+import { FontVariation } from '@harness/design-system'
+import * as Yup from 'yup'
+import { useStrings } from 'framework/strings'
+import { NameIdDescriptionTags } from '@common/components'
+import type { IacResourceStack, StackWizardStepProps } from '.'
+import css from './StackWizard.module.scss'
+
+type Provisioner = {
+  name: string
+  icon: IconName
+  value: string
+}
+
+const provisioners: Provisioner[] = [
+  {
+    name: 'Terrarform',
+    icon: 'service-terraform',
+    value: 'terraform'
+  },
+  {
+    name: 'Terragrunt',
+    icon: 'service-terraform',
+    value: 'terragrunt'
+  },
+  {
+    name: 'Pulumi',
+    icon: 'service-terraform',
+    value: 'pulumi'
+  },
+  {
+    name: 'CDK',
+    icon: 'service-terraform',
+    value: 'cdk'
+  },
+  {
+    name: 'Ansible',
+    icon: 'service-terraform',
+    value: 'ansible'
+  },
+  {
+    name: 'Cloud Formation',
+    icon: 'service-terraform',
+    value: 'cloudformation'
+  }
+]
+
+type ProvisionerTypeOption = {
+  label: string
+  icon: IconName
+  value: string
+}
+
+const ProvisionerTypeStep: React.FC<StackWizardStepProps> = props => {
+  const { name, identifier, nextStep, prevStepData } = props
+  const { getString } = useStrings()
+  const provisionerTypeOptions = useMemo(
+    (): ProvisionerTypeOption[] =>
+      provisioners.map((type: Provisioner) => ({
```

Comment:
You probably don't need to tell TS that this is a `Provisioner`, it can probably get that from the fact that `provisioners` is an array of them.
```suggestion
      provisioners.map(type => ({
```